src/planscape/planning/views.py

```diff
@@ -305,13 +365,74 @@ def create_scenario(request: HttpRequest) -> HttpResponse:
             scenario=scenario)
         scenario_result.save()
 
-        return HttpResponse(str(scenario.pk))
+        return HttpResponse(
+            json.dumps({'id': scenario.pk}),
+            content_type="application/json")
     except Exception as e:
         return HttpResponseBadRequest("Ill-formed request: " + str(e))
 
+
 #TODO - when we want to support multiple scenario results for the same scenario:
 #def create_result_for_scenario(request: HttpRequest) -> HttpResponse:
 #def list_results_for_scenario(request: HttpRequest) -> HttpResponse:
+#def get_latest_result_for_scenario
+
+
+
+def update_scenario(request: HttpRequest) -> HttpResponse:
+    """
+    Updates a scenario's name or notes.  To date, these are the only fields that
+    can be modified after a scenario is created.  This can be also used to clear
+    the notes field, but the name needs to be defined always.
+
+    Calling this without anything to update will not throw an error.
+
+    Requires a logged in user.  Users can modify only their owned scenarios.
+
+    Returns: id: The scenario's ID, even if nothing needed updating.
+
+    Required params:
+      id (int): ID of the scenario to retrieve.
+    """
+    try:
+        user = _get_user(request)
+        if user is None:
+            raise ValueError("User must be logged in.")
+
+        body = json.loads(request.body)
+        scenario_id = body.get('id', None)
+        if scenario_id is None:
+            raise ValueError("Scenario ID is required.")
+
+        scenario = Scenario.objects.select_related('planning_area__user').get(id=scenario_id)
+        if (scenario.planning_area.user.pk != user.pk):
+            # This matches the same error string if the planning area doesn't exist in the DB for any user.
+            raise ValueError("Scenario matching query does not exist.")
+
+        is_dirty = False
+        
+        if 'notes' in body:
+            # This can clear the notes field
+            scenario.notes = body.get('notes')
+            is_dirty = True
+
+        if 'name' in body:
```

Comment:
Same future note as the one for planning area: Will need to eventually make sure that the user can't update the Scenario name to be identical to another scenario under the same planning area

Reply:
Same here as the other note - the unique constraint works here.